src/Arcus.Messaging.Tests.Runtimes.AzureFunction.ServiceBus.Queue/Functions/OrderProcessingFunction.cs

```diff
@@ -31,10 +36,14 @@ public OrderProcessingFunction(IAzureServiceBusMessageRouter messageRouter)
             CancellationToken cancellationToken)
         {
             log.LogInformation($"C# ServiceBus queue trigger function processed message: {message.MessageId}");
-            
-            var context = new AzureServiceBusMessageContext(message.MessageId, _jobId, AzureServiceBusSystemProperties.CreateFrom(message), message.ApplicationProperties);
-            MessageCorrelationInfo correlationInfo = message.GetCorrelationInfo();
-            await _messageRouter.RouteMessageAsync(message, context, correlationInfo, cancellationToken);
+
+            AzureServiceBusMessageContext context = message.GetMessageContext(_jobId);
+            (string transactionId, string operationParentId) = message.ApplicationProperties.GetTraceParent();
+
+            using (var result = MessageCorrelationResult.Create(_telemetryClient, transactionId, operationParentId))
+            {
+                await _messageRouter.RouteMessageAsync(message, context, result.CorrelationInfo, cancellationToken);
+            }
```

Comment:
Will be migrated towards an isolated environment in https://github.com/arcus-azure/arcus.messaging/issues/346